game/src/ai.ts

```diff
@@ -0,0 +1,99 @@
+import { Game } from "./game";
+import { GameState, Move } from "./types";
+
+function calculateBestMove(
+  game: Game,
+  state: GameState,
+  depth: Number,
+  baseMove: Move,
+): Move {
+  let deck = state.decks[game.state.currentPlayer];
+
+  let bestMove = {
+    player: game.state.currentPlayer,
+    cardId: null,
+    placement: { row: 0, col: 0 },
+  };
+  let bestScore = -99;
+
+  if (baseMove) {
+    bestMove = baseMove;
+  }
+
+  // goes over all the deck cards
+  for (let deckCard of deck.cells) {
+    if (deckCard.cardId) {
+      let card = game.cards[deckCard.cardId];
+
+      if (typeof card.value === "number") {
+        // checks the entire board to see if it can be placed
+        for (let x = 0; x < game.numRows; x++) {
+          for (let y = 0; y < game.numCols; y++) {
```

Comment:
Try not to mix up `col, row` and `x, y` conventions.

Case in point, `row` is actually the `y` coordinate (rows are stacked vertically), whereas `col` is `x`. If we keep `col, row` as board coordinates and `x, y` as pixel coordinates (for PIXI graphics), then we never have that kind of confusion.